lib/rouge/lexers/livescript.rb

```diff
@@ -0,0 +1,305 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Livescript < RegexLexer
+      tag 'livescript'
+      aliases 'ls'
+      filenames '*.ls'
+      mimetypes 'text/livescript'
+
+      title 'LiveScript'
+      desc 'LiveScript, a language which compiles to JavaScript'
+
+      def self.detect?(text)
+        return text.shebang? 'lsc'
+      end
+
+      def self.declarations
+        @declarations ||= Set.new %w(const let var function class extends implements)
+      end
+
+      def self.keywords
+        @keywords ||= Set.new %w(
+          loop until for in of while break return continue switch case
+          fallthrough default otherwise when then if unless else throw try
+          catch finally new delete typeof instanceof super by from to til
+          with require do debugger import export yield
+        )
+      end
+
+      def self.constants
+        @constants ||= Javascript.constants + %w(yes no on off void)
+      end
+
+      def self.builtins
+        @builtins ||= Javascript.builtins + %w(this it that arguments)
+      end
+
+      loop_control_keywords = Set.new %w(break continue)
+      id = /[$a-z_]((-(?=[a-z]))?[a-z0-9_])*/i
+      int_number = /\d+(_\d+)*/
+      int = /#{int_number}(e-?#{int_number})?([A-Za-z_][A-Za-z0-9_]*)?/
```

Comment:
What's the reason for the `([A-Za-z_][A-Za-z0-9_]*)?` component of the regex? How would that be used?

Reply:
In Livescript you can group digits using underscore as a separator and you can postfix numbers with letters to describe units: http://livescript.net/#literals. Additional underscores and digits are also ignored by the transpiler so they can be part of units.

The part with `e` should be changed to also accept `+`:
```suggestion
      int = /#{int_number}(e[+-]?#{int_number})?([A-Za-z_][A-Za-z0-9_]*)?/
```